### PR TITLE
Reduce GitHub requests count

### PR DESCRIPTION
**Pull request by olmisa**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Every time when Unity editor enters in play mode the extension tries to check the latest version. When the editor enters in play mode then the last version check date was always 'DateTime.MinDate'. Added filling this information from cached json data.

## Misc

AB#73973

### CODE REVIEW

File: Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs [hunk 1]
```diff
@@ -11,20 +12,35 @@ namespace AppCenterEditor
...
         private static AppCenterEditorPrefsSO _instance;
 
-        public string EdSet_latestSdkVersion { get { return _latestSdkVersion; } set { _latestSdkVersion = value; _lastSdkVersionCheck = DateTime.UtcNow; } }
-        public string EdSet_latestEdExVersion { get { return _latestEdExVersion; } set { _latestEdExVersion = value; _lastEdExVersionCheck = DateTime.UtcNow; } }
-        public DateTime EdSet_lastSdkVersionCheck { get { return _lastSdkVersionCheck; } }
-        public DateTime EdSet_lastEdExVersionCheck { get { return _lastEdExVersionCheck; } }
+        public string EdSet_latestSdkVersion
+        {
+            get { return _latestSdkVersion; }
+            set { _latestSdkVersion = value; _lastSdkVersionCheck = DateTime.UtcNow; _lastSdkVersionCheckDateString = _lastSdkVersionCheck.ToString(CultureInfo.InvariantCulture); }
```
Comment (annakocheshkova) [inline Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs]: That's not very good code style :smile:

Comment (annakocheshkova) [reply Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs]: Please separate into multiple lines

**Review by annakocheshkova — DISMISSED** [review]

File: Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs (hunk 1 above)
Comment (olmisa) [reply Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs]: Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # App Center SDK for Unity Editor Extensions Change Log
 
+## Development version
+
+* **[Bug fix]** Fix call to github on every time when editor enters play mode.
```
Comment (guperrot) [inline CHANGELOG.md]: Can we rephrase this? Not sure to understand the wording.

Comment (olmisa) [reply CHANGELOG.md]: Fixed

**Review by annakocheshkova — DISMISSED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # App Center SDK for Unity Editor Extensions Change Log
 
+## Development version
+
+* **[Bug fix]** Fixed excess GitHub API calls from unity extension
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
* **[Bug fix]** Fixed excess GitHub API calls from unity extension.
```

Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
* **[Bug fix]** Fixed excessive amount of GitHub API calls from unity extension.
```

**Review by annakocheshkova — DISMISSED** [review]

**Review by ilyashumihin — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # App Center SDK for Unity Editor Extensions Change Log
 
+## Development version
+
+* **[Bug fix]** Fixed excessive amount of GitHub API calls from unity extension.
```
Comment (MatkovIvan) [inline CHANGELOG.md]: This is "unity extension" repo, may be just "the extension"?

**Review by russelarms — DISMISSED** [review]

**Review by Jamminroot — DISMISSED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (guperrot) [reply CHANGELOG.md]:
```suggestion
* **[Bug fix]** Fixed excessive amount of GitHub API calls.
```

File: Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs [hunk 5]
```diff
@@ -11,6 +12,8 @@ namespace AppCenterEditor
 #endif
     public class AppCenterEditorPrefsSO : ScriptableObject
     {
+        public const string EdxLastCheckDateKey = "EdxLastCheckDateKey";
```
Comment (jaeklim) [inline Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs]: Edx or EdEx?

Comment (ilyashumihin) [reply Source/Assets/AppCenterEditorExtensions/Editor/Scripts/Utils/AppCenterEditorPrefsSO.cs]: Renamed the variable to EdEx.

**Review by MatkovIvan — APPROVED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Review by guperrot — APPROVED** [review]

**Pull request merged**